crates/prql-compiler/src/semantic/resolver/transforms.rs

```diff
@@ -553,7 +552,7 @@ impl TransformCall {
                 // pipeline's body is resolved, just use its type
                 let Func { body, .. } = pipeline.kind.as_func().unwrap().as_ref();
 
-                body.lineage.clone().unwrap()
+                ty_frame_or_default(body)?
```

Comment:
Given how bad the error message is, let's just point to the issue? 

```suggestion
                ty_frame_or_default(body).map_err(anyhow!("Invalid lineage. See #3266 for more details."))?
```